Prototope/Layer.swift

```diff
@@ -637,52 +760,72 @@ public class Layer: Equatable, Hashable {
 
 	private func imageDidChange() {
 		if let image = image {
-			imageView?.image = image.uiImage
+			imageView?.image = image.systemImage
 			size = image.size
 			layer.masksToBounds = self._shouldMaskToBounds()
 		}
 	}
 
-	private init(wrappingView: UIView, name: String? = nil) {
+	private init(wrappingView: SystemView, name: String? = nil) {
 		view = wrappingView
 		self.name = name
 	}
     
     
 	/** Creates a new layer hosted by the given view. The layer wraps its own view, which is sized to the full dimensions of the hosting view. */
-    convenience init(hostingView: UIView, name: String? = nil) {
+    convenience init(hostingView: SystemView, name: String? = nil) {
         self.init()
         self.parentView = hostingView
 		self.frame = Rect(hostingView.bounds)
-		self.view.autoresizingMask = .FlexibleWidth | .FlexibleHeight
+		#if os(iOS)
+			self.view.autoresizingMask = .FlexibleWidth | .FlexibleHeight
+		#else
+			self.view.autoresizingMask = NSAutoresizingMaskOptions.ViewWidthSizable | NSAutoresizingMaskOptions.ViewHeightSizable
+			
+		#endif
     }
 
 	// MARK: UIKit mapping
 
-	var view: UIView
-	private var layer: CALayer { return view.layer }
+	var view: SystemView
+	private var layer: CALayer {
+		#if os(iOS)
+		return view.layer
+		#else
+		return view.layer!
+		#endif
+	}
 	private var imageView: TouchForwardingImageView? { return view as? TouchForwardingImageView }
 
-	private var parentView: UIView? {
+	private var parentView: SystemView? {
 		get { return view.superview }
 		set { newValue?.addSubview(view) }
 	}
 
 	// MARK: Touch handling implementation
 
-	class TouchForwardingImageView: UIImageView {
+	class TouchForwardingImageView: SystemImageView {
 		required init(coder aDecoder: NSCoder) {
 			fatalError("This method intentionally not implemented.")
 		}
 
 		override init(frame: CGRect) {
 			super.init(frame: frame)
+			#if os(OSX)
+				self.wantsLayer = true
+				
+				// TODO(jb): Can probably lazily add this when a mouse Entered/moved/exited event happens, so not all layers need to have tracking areas by default.
```

Comment:
What about adding this when a `mouseMovedHandler` is set? My understanding is that these are quite expensive.